crates/shadowsocks/src/relay/socks5.rs

```diff
@@ -213,6 +210,47 @@ pub enum Address {
 }
 
 impl Address {
+    /// read from a cursor
+    pub fn read_cursor<T: AsRef<[u8]>>(cur: &mut io::Cursor<T>) -> Result<Address, Error> {
```

Comment:
You just made a new function `Address::read_cursor`, then it will have exactly the same code as the `Address::read_from`. Is it possible to unify these two functions?

Reply:
tbh i'm not sure - the logic are same, but reusing the sync and async calls seems non trivial. you got any suggestions?

Reply:
You are right. I am ok with it.